clientconn.go

```diff
@@ -461,15 +461,15 @@ var _ ClientConnInterface = (*ClientConn)(nil)
 // handshakes. It also handles errors on established connections by
 // re-resolving the name and reconnecting.
 type ClientConn struct {
-	ctx    context.Context
-	cancel context.CancelFunc
+	ctx    context.Context    // Initialized using the background context at dial time.
+	cancel context.CancelFunc // Cancelled on close.
 
 	// The following are initialized at dial time, and are read-only after that.
 	target            string                // User's dial target.
 	parsedTarget      resolver.Target       // See parseTargetAndFindResolver().
 	authority         string                // See determineAuthority().
 	dopts             dialOptions           // Default and user specified dial options.
-	balancerBuildOpts balancer.BuildOptions // Options passed to balancer at build time.
+	balancerBuildOpts balancer.BuildOptions // Balancer build options, sourced from user provided options to dial.
```

Comment:
This seems misleading to me, after looking it up.  I see now that it comes from many different places.  But I'm wondering why it's a field at all.  Maybe we should just delete it and create when we create the balancer?  Also is it possible it's wrong to do it this way, because we clone the transport creds, but we only do that cloning once at init time?  Maybe `// <no explanation needed> TODO: delete`?

Reply:
Yes, I think we can get rid of this field once we switch to the `gsb` because we pass these options to `gracefulswitch.NewBalancer()` at dial time. So, added a TODO to delete this field.